pkg/fileutil/preallocate_darwin.go

```diff
@@ -30,14 +30,28 @@ func preallocExtend(f *os.File, sizeInBytes int64) error {
 }
 
 func preallocFixed(f *os.File, sizeInBytes int64) error {
+	// allocate contiguous space on disk
 	fstore := &syscall.Fstore_t{
-		Flags:   syscall.F_ALLOCATEALL,
+		Flags:   syscall.F_ALLOCATECONTIG,
```

Comment:
i do not think it worth the effort to optimize this on macos. people will not run etcd in production on osx anyway.